File: SupplyFlowManagerAPI/src/main/java/com/projedata/SupplyFlow/Manager/entity/ProductComposition.java
package com.projedata.SupplyFlow.Manager.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "product_composition")
public class ProductComposition {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long code;

    private Double quantityRequired;

    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false)
    @JsonBackReference
    private Product product;

    @ManyToOne
    @JoinColumn(name = "raw_material_id", nullable = false)
    private RawMaterial rawMaterial;

}

File: SupplyFlowManagerFront/src/types/ProductComposition.ts
import type { UnitOfMeasure } from "@/types/RawMaterial";

export interface ProductCompositionItem {
  id?: number;
  productId: number;
  rawMaterialId: number;
  rawMaterialName: string;
  quantityRequired: number;
  unitOfMeasure: UnitOfMeasure;
}

export interface SaveProductCompositionPayload {
  productId: number;
  rawMaterialId: number;
  quantityRequired: number;
}
